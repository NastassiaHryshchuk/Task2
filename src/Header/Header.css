  .header {
    padding-top: 10px;
    padding-bottom: 20px;
    background-color: rgb(53, 53, 53);
  }

  h2 {
    color: rgb(255, 255, 255);
    text-transform: uppercase;
  }

  input {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    box-sizing: border-box;
    background-color: rgb(22, 18, 18);
    border: none; 
    border-bottom: solid 2px  rgb(247, 66, 96);
    color: rgb(255, 255, 255);
    margin-bottom: 20px;
  }

  input:focus {
    outline: none;
  }

  .span-search {
    color: rgb(255, 255, 255);
    text-transform: uppercase;
  }

  .container {
    padding: 20px 100px;
  }

  .flex-container {
    display: flex;
  }

  .flex-left {
    margin-left: auto;
  }

  .button-search {
    margin-left: auto;
    padding: 5px 55px;
    border: none;
    color: rgb(255, 255, 255);
    border-radius: 3px;
    background-color: rgb(247, 66, 96);
    text-transform: uppercase;
    cursor: pointer;
  }

  .button-search:hover, 
  .button-search:focus,
  .button-search:active  {
    background-color: rgb(128, 128, 128);
  }

  .button-title, 
  .button-director {
    margin-left: 20px;
    padding: 5px 12px;
    border: none;
    color: rgb(255, 255, 255);
    border-radius: 3px;
    cursor: pointer;
  }

  .button-title {
    text-transform: uppercase;
    background-color: rgb(247, 66, 96);
  }

  .button-title:hover, 
  .button-title:focus,
  .button-title:active  {
    background-color: rgb(128, 128, 128);
  }

  .button-director {
    text-transform: uppercase;
    background-color: rgb(128, 128, 128);
  }

  .button-director:hover, 
  .button-director:focus,
  .button-director:active  {
    background-color: rgb(128, 128, 128);
  }
  

 .results-count, 
 .results-filter {
    font-size: 13px;
  }

  .button-sub-header {
    background: none;
    outline: none;
    border: none;
    margin-left: 20px;
    cursor: pointer;
  }

  .button-sub-header:hover, 
  .button-sub-header:focus,
  .button-sub-header:active {
    color: rgb(247, 66, 96);
  }
