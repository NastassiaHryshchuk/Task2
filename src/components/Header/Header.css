.header {
  padding-top: 10px;
  padding-bottom: 20px;
  background-color: rgb(53, 53, 53);
  background-image: url(../../assets/background.jpg);
}

.wrap {
  width: 80%;
  margin: 0 auto;
}

h2 {
  color: rgb(255, 255, 255);
  text-transform: uppercase;
}

input {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
  background-color: rgb(22, 18, 18);
  border: none;
  border-bottom: solid 2px rgb(247, 66, 96);
  color: rgb(255, 255, 255);
  margin-bottom: 20px;
}

input:focus {
  outline: none;
}

.span_search {
  color: rgb(255, 255, 255);
  text-transform: uppercase;
}

.flex_container {
  display: flex;
}

.button_search {
  margin-left: auto;
  padding: 5px 55px;
  border: none;
  color: rgb(255, 255, 255);
  border-radius: 3px;
  background-color: rgb(247, 66, 96);
  text-transform: uppercase;
  cursor: pointer;
}

.button_search:hover,
.button_search:focus,
.button_search:active {
  background-color: rgb(128, 128, 128);
}

.btn {
  margin-left: 20px;
  padding: 5px 12px;
  border: none;
  color: rgb(255, 255, 255);
  background-color: rgb(128, 128, 128);
  border-radius: 3px;
  cursor: pointer;
  text-transform: uppercase;
}

.toggle {
  display: none;
}

.toggle:checked + label {
  background-color: rgb(247, 66, 96);
}
