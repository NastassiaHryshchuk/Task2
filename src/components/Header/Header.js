import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import Search from '../../components/Search/Search';
import Detail from '../Detail/Detail';

class Header extends Component {
  render() {
    if (!this.props.crt) {
      return <Search />;
    }
    return <Route path="/film/:id" component={Detail} />;
  }
}

const mapStateToProps = state => {
  return {
    crt: state.currentFilm,
  };
};

export default connect(mapStateToProps)(Header);

// <Route path="/film/:id" component={Detail} />;
