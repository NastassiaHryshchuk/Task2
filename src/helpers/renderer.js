import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Routes from '../Routes/Routes';

export default(req, store) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.path} context={{}}>
        <Routes />
      </StaticRouter>
    </Provider>,
  );

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <title>SSR</title>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.INITIAL_STATE = ${JSON.stringify(store.getState())}
        </script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
};
