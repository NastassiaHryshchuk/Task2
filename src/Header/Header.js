import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
  render() {
    return (
        <header className="header">
            <div className="wrap">
                <span className="logo">netflixroulette</span>
                <h2>Find your move</h2>
                 <input type="text" value="Quentin Tarantino" />
                <div className="flex-container">
                    <span className="span-search">Search by</span>
                    <button className="button-title">Title</button>
                    <button className="button-director">Director</button>
                    <button className="flex-left button-search">Search</button>
                </div>
            </div >
      </header>
    );
  }
}

export default Header;
