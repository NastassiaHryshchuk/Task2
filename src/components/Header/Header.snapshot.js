import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <Header />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
