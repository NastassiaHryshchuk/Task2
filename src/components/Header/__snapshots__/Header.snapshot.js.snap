// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot renders 1`] = `
<header
  className="header"
>
  <div
    className="wrap"
  >
    <a
      className="logo"
      href="#"
    >
      netflixroulette
    </a>
    <h2>
      Find your move
    </h2>
    <form>
      <input
        placeholder="Search here..."
        required={true}
        type="text"
      />
      <div
        className="flex_container"
      >
        <span
          className="span_search"
        >
          Search by
        </span>
        <input
          checked={true}
          className="toggle"
          id="titleChoice"
          name="film"
          type="radio"
          value="title"
        />
        <label
          className="btn"
          htmlFor="titleChoice"
        >
          Title
        </label>
        <input
          className="toggle"
          id="directorChoice"
          name="film"
          type="radio"
          value="director"
        />
        <label
          className="btn"
          htmlFor="directorChoice"
        >
          Director
        </label>
        <button
          className="button_search"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</header>
`;
